scripts/build.py

```diff
@@ -206,6 +206,15 @@ def generate_app_yaml(deploy_mode=False, maintenance_mode=False):
     # is being run from the deploy script.
     if deploy_mode:
         content = content.replace('version: default', '')
+        # The DATASTORE_* environment variables are only needed to test locally,
```

Comment:
I'm not entirely sure why this is here. Is it just as a safeguard? This PR doesn't seem to change app_dev.yaml...

Reply:
Yes it's a safe guard. Adding the environment variables to app_dev.yaml in this PR would break the server because Cloud NDB is not a requirement yet.

Reply:
OK, I suggest adding a longer note to explain all the context, otherwise this is quite mysterious given that grepping DATASTORE_DATASET etc. doesn't lead to any other references (so it looks like dead code).

Reply:
OK well, since the requirements of this context has changed (only needed for backend tests), I've moved all env variable management into the `managed_cloud_datastore_emulator` function itself. Resolving this as obsolete.